### DISCUSSION TITLE
Implementing syntax highlighting

### DISCUSSION THREAD

**Discussion by VideoCarp** [emilk/egui#1628] [Q&A]
I'm writing a text editor and trying to implement syntax highlighting,
so I followed the example in [the demo](https://egui.rs). It works there, so I thought it might be useful. But I am getting an error that `syntax_highlighting` is not found in the crate root. I have syntect and egui both installed at their latest versions and have them
as dependencies.
```rust
            let mut theme = crate::syntax_highlighting::CodeTheme::from_memory(ui.ctx());
            ui.collapsing("Theme", |ui| {
                ui.group(|ui| {
                    theme.ui(ui);
                    theme.clone().store_in_memory(ui.ctx());
                });
            });


            let mut layouter = |ui: &egui::Ui, string: &str, wrap_width: f32| {
                let mut layout_job = 
                    crate::syntax_highlighting::highlight(ui.ctx(), &theme, string, self.language);
                layout_job.wrap.max_width = wrap_width;
                ui.fonts().layout_job(layout_job)
            };
```

In particular, it is these lines that raise the error that `syntax_highlighting` is not found in the crate root. In case it's useful, here is all my code:
https://pastebin.com/mHDhLhSR
(do ignore the errors with `ScrollArea`, I have fixed those now.)

**I want to implement syntax highlighting to my egui application, but I get the error that `syntax_highlighting` is not found in the crate root, which the only example I found used.**

**Comment by Mingun** [comment]
Just remove the `crate::` part from the path. `crate::` means a module defined in your crate, but `syntax_highlighting` is a crate name (you also could think about it as a root module name of that crate)

**Reply by VideoCarp** [reply]
Errors with ``use of undeclared crate or module `syntax_highlighting` ``
My cargo.toml: https://paste.gg/p/anonymous/52c35f32ba4b4a9cad82919f9fb0e7e0
In case it's relevant.

**Reply by Mingun** [reply]
Ah... `syntax_highlighting` is the module defined in the demo [here](https://github.com/emilk/egui/blob/master/egui_demo_lib/src/syntax_highlighting.rs). You need to copy it to your project (but notice, that this code is licensed under MIT OR Apache-2.0, which restricts your ability to close your code).

**Reply by VideoCarp** [reply]
When I copied it and put it as a `mod`, it came up with 17 errors.
